tests/testthat/test-derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,183 @@
+library(tibble)
+data <- tribble(
+  ~USUBJID, ~AVISITN, ~AVALC,
+  "1",      1,        "PR",
```

Comment:
could you add a patient where we can see impact of `join_type = "at_after"`, e.g. a patient who has a CR PR PR and then the CR gets ignored, and only the first PR gets flagged

Reply:
I have updated the example (`join_type` and `filter`). Patient 4 is now showing the impact of `join_type`.

Reply:
what about adding the following to really exemplify this:
```
    "5",      1,        "PR",   NA_character_,
    "5",      1,        "CR",   NA_character_
```
As it shows you need to be after (in the unlikely event you had 2 responses at same time - but in reality they'd use date as well here instead of seq only anyway).

Reply:
If I add this patient, the order is not well defined and the function issues a warning. Thus I would not add it.

Reply:
ah ok, i wonder if the `at_before` and `at_after` options are even ever needed then? when would you have the scenario to use these if it fires a warning anyway if you had 2 records with matching order variables? might be simpler just to have `after`, `before` and `all` options

Reply:
The warning is not caused by `at_before` or `at_after`. You always get it if the variables specified for `by_vars` and `order` are not a unique key. In this case the selections specified by `join_type` and `first_cond` are questionable. The user can get rid of the warning by updating `order`, `by_vars`, or `check_type`. It depends on the data and the filter conditions which of the arguments should be updated.
I think in most cases `after`, `before`, or `all` are used but sometimes `at_before` or `at_after` may be useful as well (like in test 6).

Reply:
see my other comment - i'm strongly in favour of removing the `at_*` options. if you disagree we could add to the Q&A meet Weds and see what others feel.

Reply:
See my answer to the other comment.